src/FileSystemPolicy.cpp

```diff
@@ -182,3 +183,87 @@ long FileSystemPolicy::getFileSize(const struct stat& fileStat) const
 		return 0;
 	return fileStat.st_size;
 }
+
+/**
+ * @brief Deletes a file at the specified path.
+ *
+ * This function attempts to delete the file located at the given path.
+ * If the file cannot be deleted, it throws a runtime error with the
+ * appropriate error message.
+ *
+ * @param path The path to the file to be deleted.
+ * @throws std::runtime_error if the file cannot be deleted.
+ */
+void FileSystemPolicy::deleteFile(const std::string& path) const
+{
+	errno = 0;
+	if (remove(path.c_str()) != 0)
+		throw std::runtime_error("remove(): " + std::string(strerror(errno)));
+}
+
+/**
+ * @brief Deletes a directory at the specified path.
+ *
+ * This function attempts to delete the directory located at the given path.
+ * It recursively deletes all files and subdirectories within the directory.
+ * If any file or directory cannot be deleted, it throws a runtime error with
+ * the appropriate error message.
+ *
+ * @param path The path to the directory to be deleted.
+ * @throws std::runtime_error if any file or directory cannot be deleted.
+ */
+// NOLINTNEXTLINE (misc-no-recursion): recursion is being handled
+void FileSystemPolicy::deleteDirectory(const std::string& path) const
+{
+	DirectoryGuard dir(openDirectory(path));
+
+	struct dirent* entry = NULL;
+	while ((entry = readdir(dir.getDir())) != NULL) {
+		// Skip `.` and `..` entries
+		if (strcmp(entry->d_name, ".") == 0 || strcmp(entry->d_name, "..") == 0)
+			continue;
+
+		std::string fullPath = std::string(path) + "/" + entry->d_name;
+
+		struct stat info = getFileStat(fullPath);
+		// NOLINTNEXTLINE: misinterpretation by HIC++ standard
+		if (S_ISDIR(info.st_mode))
+			deleteDirectory(fullPath);
+		else {
+			errno = 0;
+			if (unlink(fullPath.c_str()) != 0)
+				throw std::runtime_error("unlink(): " + std::string(strerror(errno)));
+		}
+	}
+
+	// Remove the empty directory itself
+	errno = 0;
+	if (rmdir(path.c_str()) != 0)
```

Comment:
We're not allowed to use rmdir. But std::remove() also works so also deleteFile

Reply:
Changed

Reply:
Ah no, actually std::remove only deletes directories starting with C++17, so we don't rly have another choice but to use rmdir. Any suggestions?

Reply:
In that case I think we simply can't delete a directory? I didn't find any alternative, and since `rmdir` is not allowed function we simply can't / are allowed to do it?

We could check if it's a regular file and only delete in that case. Else return error 400 or maybe 403?
Or we ask for clarification in discord, but I think the answer will be: "look at list".

Reply:
I agree that asking on Discord would be fruitless. I can just set an error in the switch case, but it is rly a shame to leave it out just because of such senseless restrictions. It works wonderfully as such. In that case the bug about removing the dots in the target resource would not be relevant anymore, no?

Reply:
If the product owner say no we have to be obedient ;)

No dot removal is still necessary since you could still GET a sensitive file of the filesystem and hack the server

Reply:
Oh well, will set the error, probably Forbidden. And comment out the deleteDirectory(). But once the eval is done I'm activating it again!

Reply:
Done